pyomo/solvers/tests/mip/test_factory.py

```diff
@@ -36,23 +36,23 @@ def setUpModule():
 def tearDownModule():
     TempfileManager.tempdir = old_tempdir
 
-class TestWriter(AbstractProblemWriter):
+class DummyWriter(AbstractProblemWriter):
```

Comment:
Just a suggestion: I think it's more common to refer to these as "mock" objects in Python. I'm fine with `DummyWriter` but we may want to consider `MockWriter`